HealthcareSystem/PatientWebApp/wwwroot/js/patient_examinations.js

```diff
@@ -400,6 +565,31 @@ function third_step_next() {
         return;
     }
 
+    var stepEvent = {
+        "startSchedulingEventId": parseInt(startEventId),
+        "userAge": parseInt(age),
+        "userGender": parseInt(gender),
+        "eventStep": parseInt(2),
+        "clickEvent": parseInt(1)
+    };
+
+    $.ajax({
```

Comment:
You should've probably made separate functions for these (like, sendStartEvent, sendEndEvent, sendStepEvent instead of copying the same AJAX calls so many times.